src/natcap/invest/validation.py

```diff
@@ -596,17 +600,58 @@ def check_spatial_overlap(spatial_filepaths_list,
     return None
 
 
+def timeout(func, *args, timeout=5, **kwargs):
+    """Stop a function after a given amount of time.
+
+    Args:
+        func (function): function to apply the timeout to
+        args: arguments to pass to the function
+        timeout (number): how many seconds to allow the function to run.
+            Defaults to 5.
+
+    Returns:
+        None
+
+    Raises:
+    """
+
+
+    # use a queue to share the return value from the file checking thread
+    # the target function puts the return value from `func` into shared memory
+    message_queue = queue.Queue()
+    def wrapper_func():
+        message_queue.put(func(*args, **kwargs))
+
+    thread = threading.Thread(target=wrapper_func)
+    LOGGER.info(f'Starting file checking thread with timeout={timeout}')
+    thread.start()
+    thread.join(timeout=timeout)
+
+    if thread.is_alive():
+        return ('Validation of this file timed out. If this file is stored in '
```

Comment:
@phargogh and I discussed that it might be better to raise an error on timeout and exit validation entirely. I first tried it out this way, just returning the timeout message like any other validation warning message, because it took the least modification to things outside of `timeout()`. But this could maybe be a good use for a custom exception?

Reply:
I also thing there's a `TimeoutError` that could be appropriate as opposed to a custom exception.

Reply:
I do like the idea of not holding the user hostage if this timeouts. Meaning, on a given timeout I as the user should still be able to click run. Is that what happens in this case?

Reply:
It would still block you from running the model, that input would show up with a red X. That is interesting though, should it be possible to run the model anyway?

I kind of like this approach because it ensures that the UI will show up. Apparently if there are saved inputs from a previous run, validation happens before the model window opens. So if one of the inputs needs to be downloaded it could appear to hang while that happens. I like that this way the error message shows up in the UI. If it raised and exception instead it would have to show up in the terminal window instead and the user might not notice it.

Reply:
For future us, we were chatting about this on our internal slack channel and decided that an inability to run validation shouldn't prevent an informed user from running a model.  Rather, we'll want to display some sort of warning about the inability to validate and let the user make an informed decision.

In the longer-run, this will involve reworking the validation API to support status levels.